component/src/main/java/org/wso2/siddhi/runner/beam/SiddhiAppContainer.java

```diff
@@ -0,0 +1,287 @@
+import org.wso2.siddhi.runner.beam.extension.BeamParDoProcessor;
+import org.wso2.siddhi.runner.beam.extension.BeamSinkProcessor;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * This is where Siddhi query is generated depending on {@link DirectGraph} which in turn is created
+ * depending on structure of {@link org.apache.beam.sdk.Pipeline}.
+ */
+public class SiddhiAppContainer {
+
+    private static final Logger log = LoggerFactory.getLogger(SiddhiAppContainer.class);
+    private SiddhiAppRuntime runtime;
+//    private CommittedBundle bundle;
+    private List<String> streamDefinitions = new LinkedList<>();
+    private List<String> queryDefinitions = new LinkedList<>();
+    private Map<String, AppliedPTransform> transformsMap = new HashMap<>();
+    private Map<String, PCollection> collectionsMap = new HashMap<>();
+    private Map<String, TupleTag> additionalOutputTags = new HashMap<>();
+    private DirectGraph graph;
+    private int multiSinkcount = 1;
+
+    public SiddhiAppContainer() { }
+
+    public void createSiddhiQuery() {
+        log.info("Creating Siddhi Query");
+        ExecutionContext context = ExecutionContext.getInstance();
+        this.graph = context.getGraph();
+        for (CommittedBundle rootBundle: context.getRootBundles()) {
+            for (AppliedPTransform transform: graph.getPerElementConsumers(rootBundle.getPCollection())) {
+                if (SiddhiAppContainer.compatibleTransform(transform)) {
+                    generateSiddhiQueryForTransform(transform, rootBundle.getPCollection());
+                } else {
+                    log.error("Siddhi does not support " + transform.getTransform().toString() + " at the moment");
+                }
+            }
+        }
+    }
+
+    public void createSiddhiRuntime() throws SiddhiParserException {
+        try {
+            log.info("Creating Siddhi Runtime");
+            SiddhiManager siddhiManager = new SiddhiManager();
+            StringBuilder siddhiApp = new StringBuilder();
+            this.streamDefinitions.forEach(siddhiApp::append);
+            this.queryDefinitions.forEach(siddhiApp::append);
+            log.debug(siddhiApp.toString());
+            siddhiManager.setExtension("beam:pardo", BeamParDoProcessor.class);
+            siddhiManager.setExtension("beam:groupbykey", BeamGroupByKeyProcessor.class);
+            siddhiManager.setExtension("beam:sink", BeamSinkProcessor.class);
+            this.runtime = siddhiManager.createSiddhiAppRuntime(siddhiApp.toString());
+            this.runtime.start();
+        } catch (SiddhiParserException e) {
+            log.error("There is an error in your SiddhiQL", e.getMessage(), e);
+            throw e;
+        }
+    }
+
+    private void generateSiddhiQueryForTransform(AppliedPTransform transform, PCollection keyCollection) {
+        /*
```

Comment:
Use a single line comment if it's not more than one line instead of comment blocks. Check all places